wforce/wforce.cc

```diff
@@ -460,23 +465,329 @@ unsigned int g_num_sibling_threads = WFORCE_NUM_SIBLING_THREADS;
 SodiumNonce g_sodnonce;
 std::mutex sod_mutx;
 
+void encryptMsg(const std::string& msg, std::string& packet)
+{
+    std::lock_guard<std::mutex> lock(sod_mutx);
+    packet=g_sodnonce.toString();
+    packet+=sodEncryptSym(msg, g_key, g_sodnonce);
+}
+
+bool decryptMsg(const char* buf, int len, std::string& msg)
+{
+  SodiumNonce nonce;
+  memcpy((char*)&nonce, buf, crypto_secretbox_NONCEBYTES);
```

Comment:
also if buf was std::string, you could use std::string#copy instead.

Reply:
buffer is coming directly off a socket read so I don't want to have to convert it to std::string